### ISSUE TITLE
Not iterating through epochs

### ISSUE THREAD

**Issue by krwarner** [affinelayer/pix2pix-tensorflow#86]
When running pix2pix.py in training mode exactly as the README says to with the max_epochs set to 200, when it reaches the end of the steps it doesn't iterate through to the next epoch, it just ends the training. What is going on?

**Comment by julien2512** [comment]
Do you mean it reaches the 200 epochs or it reaches the 1 one ?

**Comment by krwarner** [comment]
It only does 1 epoch.

**Comment by julien2512** [comment]
If there is nothing more in the output dir, I don't know. May be should use "trace_freq" to see more.

**Comment by krwarner** [comment]
trace_freq doesn't really help. The program just reaches the final step (step defined as number of training steps ie. --max_steps) and does not continue on beyond 1 epoch.

**Comment by julien2512** [comment]
Assuming you got no errors, it is like you did not set --max_epochs !
max_epochs do not have default value.

In the standard output header there is something like this :
> max_epochs = 600
> max_steps = None
> mode = train
(in my case, I do not set max_steps so it is deducted from max_epochs & examples count).

**Comment by krwarner** [comment]
If you look at my original post, I set max_epochs to 200, just as the README suggests and did not set a max_steps.

**Comment by julien2512** [comment]
Alright, alright ! Your parameters & standard output is ok.
I can't figure out why it is not working for you.
The only ways to stop in the code are :
> for step in range(max_steps)
and
> if sv.should_stop()
I do not know what are the conditions to fire the should_stop method.
And I believe you do not have 2**32 samples (kidding).

In my case, I always had run pix2pix inside a docker container.
I was too afraid my ubuntu not to be tensorflow complainant.
Within the container, it was working the very first time.

**Comment by krwarner** [comment]
Yeah, I have gone through the code and can't seem to understand why the should_stop() is being triggered, and yes I am not doing 2**32 samples lol. 

I have tried on a windows machine and a virtualbox running ubuntu, both have the same issue, perhaps I will try running a docker container and try again.
